### ISSUE TITLE
Consider introducing general `label` field to processors (and other components?)

### ISSUE THREAD

**Issue by Jeffail** [Jeffail/benthos#510]
Currently the labels for components is either the path (`pipeline.processors.0.1.processors.0`) of the component, or if it's a resource the name (`resources.processors.foo`). This means if a user wants nice labels in their logs and metrics they need to configure them as resources.

It would be possible to add a general purpose `label` field to components that overrides the default label assigned to it which removes that restriction. However, this is likely to need breaking changes in the logging and metrics set up for components and needs some further consideration.

For users it would look something like this:

```yaml
pipeline:
  processors:
    - bloblang: 'root = this.foo.bar'
      label: 'my_foo_mapping'
```

**Comment by Jeffail** [comment]
Another smart thing to do would be to have this automatically be a label for metrics aggregators that support it, and part of the metric name for ones that don't. This would obviously also be a breaking change so would need thinking through and a V4.

**Comment by Jeffail** [comment]
Going to use this issue as a general purpose "rework metrics components" task. Other items that should be included here are:
1. The ability to have floating point gauges
2. Simplified interface for metrics implementations, labels/tags get first class support
3. The mapping field also gets first class support
4. No more obfuscated renames (statsd vs prometheus)

**Comment by Jeffail** [comment]
Done and released as of v3.43.0
